### ISSUE TITLE
Community charts suggestion: Synapse (Matrix server)

### ISSUE THREAD

**Issue by MrXavi3** [truenas/apps#167]
[Synapse](https://github.com/matrix-org/synapse) is Matrix homeserver, Matrix is an open standard for communications on the internet, supporting federation, encryption and VoIP.

**Comment by stavros-k** [comment]
I believe we should go with Dendrite https://github.com/element-hq/dendrite.
Looks like its >90% in terms of  synapse-feature parity. Maybe pair with with element-web

**Comment by PrimozR** [comment]
> I believe we should go with Dendrite https://github.com/element-hq/dendrite. Looks like its >90% in terms of synapse-feature parity. Maybe pair with with element-web

Either option I support adding a matrix server to the app lineup.

**Comment by DaBaldEagul** [comment]
Dendrite isn't in active development as of right now (it is in maintenance mode, only security updates are provided), so I do think that Synapse (or alternatively, Continuwuity or Tuwunel) would be better to support.

I'd really love it if any of these were supported though!

**Comment by PrimozR** [comment]
> Dendrite isn't in active development as of right now (it is in maintenance mode, only security updates are provided), so I do think that Synapse (or alternatively, Continuwuity or Tuwunel) would be better to support.
> 
> I'd really love it if any of these were supported though!

Correct, I only noticed that in the past few days but didn't post about it here. It has been in maintenance mode for about a year now though.
What is the general status with Matrix anyway? A few years ago they ran out of funding and had an exodus, is the company/project stable now?
What about the alternatives you mention, are they alternatives? I'll look into it now.

**Comment by DaBaldEagul** [comment]
> Correct, I only noticed that in the past few days but didn't post about it here. It has been in maintenance mode for about a year now though.

Yup, it has indeed been a while. Looking at the issues, the latest "Are We Synapse Yet" tagged issue (basically meaning major features to be implemented) was closed on November 1, 2024. More than a year ago. Sliding sync is among them, which is, for as far as I can gather, a rather important feature for better responsiveness and client support. So I don't think that Dendrite is a great option as of now.

> What is the general status with Matrix anyway? A few years ago they ran out of funding and had an exodus, is the company/project stable now? 

From what I can gather they're doing alright, though I am not heavily invested in Element (the company running most of Matrix). They seem to be getting funding through commercial customers, by offering hosting for the Synapse homeserver.

> What about the alternatives you mention, are they alternatives? I'll look into it now.

Tuwunel is listed on matrix.org as "Stable", Continuwuity as "Beta", though they seem to be nearing their first stable release. 

I believe though that there has been some drama between these two projects, with the lead developer of Tuwunel apparently sending death threats to the folks working on Continuwuity. Make of that what you will. Again I am not very invested at this time, so I cannot confirm or deny this.

**Comment by hasmar04** [comment]
Dendrite [seems to be active again](https://github.com/element-hq/dendrite), but is still marked as Beta on the [Matrix Servers List](https://matrix.org/ecosystem/servers/)

**Comment by DaBaldEagul** [comment]
For as far as I can tell it is still in maintenance mode — no new features are added, only security patches. At least, that's what the README.md (still) claims.

**Comment by hasmar04** [comment]
Ah yep I missed that part.

**Comment by HubGon** [comment]
Hi,

i am interested in this app too. Is there anything new to this?
If not, is there a favorite backend? I would like to try getting it working locally, to open a PR to one of them.

Kind regards

**Comment by PrimozR** [comment]
FYI, Continuwuity has been released as an app.
